clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -652,7 +655,7 @@ public void subscribe(List<String> topics, ConsumerRebalanceListener listener) {
         try {
             log.debug("Subscribed to topic(s): {}", Utils.join(topics, ", "));
             this.subscriptions.subscribe(topics, listener);
-            metadata.setTopics(topics);
+            metadata.setTopics(subscriptions.groupSubscription());
```

Comment:
That makes sense, could you rephrase the above a little bit as comments to groupSubscription?